common/src/main/java/com/wynntils/screens/overlays/placement/OverlayManagementScreen.java

```diff
@@ -612,6 +614,11 @@ private Pair<Double, Double> calculateDragAfterSnapping(double dragX, double dra
             return new Pair<>(dragX, dragY);
         }
 
+        dragX += snapOffsetX;
+        dragY += snapOffsetY;
+        snapOffsetX = dragX;
+        snapOffsetY = dragY;
```

Comment:
Can you just create a variable "originalDragX/Y" and set snapOffset with that before returning? This code is currently really hard to understand from looking at it.